compiler/include/symbol.h

```diff
@@ -383,6 +393,12 @@ class FnSymbol : public Symbol {
   bool            isMethod()                                   const;
   bool            isPrimaryMethod()                            const;
   bool            isSecondaryMethod()                          const;
+  bool            isIterator()                                 const;
```

Comment:
It _could_ be used elsewhere in the compiler. It's not currently, though. It is similar to isMethod(), which is why I added it and use it in FnSymbol::printDocs() and FnSymbol::docsDirective().